### ISSUE TITLE
Plugin version conflict on Logstash 2.4

### ISSUE THREAD

**Issue by ryac** [logstash-plugins/logstash-output-kafka#91]
Hi all, 

I've installed a fresh copy of Logstash 2.4 on Ubuntu 14.04. I plan to use Kafka version 0.10.0.1. According to the docs, I need to upgrade the logstash-output-kafka and logstash-input-kafka plugins to version 5.0.0.

When I try to install, I get a "plugin version conflict" error. Here's the output:

```
vagrant@kafka-server:/opt/logstash$ sudo bin/logstash-plugin install --version 5.0.0 logstash-output-kafka
Validating logstash-output-kafka-5.0.0
Installing logstash-output-kafka
Plugin version conflict, aborting
ERROR: Installation Aborted, message: Bundler could not find compatible versions for gem "logstash-core-plugin-api":
  In snapshot (Gemfile.lock):
    logstash-core-plugin-api (= 1.60.4)

  In Gemfile:
    logstash-output-udp (< 3.0.0) java depends on
      logstash-core-plugin-api (~> 1.0) java

    logstash-output-udp (< 3.0.0) java depends on
      logstash-core-plugin-api (~> 1.0) java

    logstash-output-udp (< 3.0.0) java depends on
      logstash-core-plugin-api (~> 1.0) java

<...more listings here...>

    logstash-output-udp (< 3.0.0) java depends on
      logstash-core-plugin-api (~> 1.0) java

    logstash-output-udp (< 3.0.0) java depends on
      logstash-core-plugin-api (~> 1.0) java

    logstash-output-kafka (= 5.0.0) java depends on
      logstash-core-plugin-api (~> 2.0) java

    logstash-core-plugin-api (= 1.60.4) java

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

How can I resolve this? Both kafka plugins throw the same error, is this more of a Logstash issue?

I originally posted this here:
https://discuss.elastic.co/t/plugin-version-conflict-with-logstash-2-4-logstash-input-kafka-and-logstash-output-kafka/59562

**Comment by wiibaa** [comment]
@ryac According to [Changelog](https://github.com/logstash-plugins/logstash-output-kafka/blob/master/CHANGELOG.md) this constraint was corrected in 5.0.1, can you try to update to this specific version or higher

**Comment by ryac** [comment]
Thanks, that was the issue. I was able to install both output- and input-kafka plugins, version 5.0.4.
